sum_root_to_leaf.py

```diff
@@ -0,0 +1,31 @@
+# Definition for a binary tree node.
+# class TreeNode:
+#     def __init__(self, x):
+#         self.val = x
+#         self.left = None
+#         self.right = None
+
+class Solution:
+    
+    def sumNumbers(self, root: TreeNode) -> int:
+    
+	# Check input
+	    if not root:
+		    return 0
+
+	    result = []
+	    self.dfs(root, result, "")
+	    return sum([int(i) for i in result])
+
+    def dfs(self, root, result, num):
+        
+	    if not root.left and not root.right:
+		    result.append(num + str(root.val))
+	    if root.left:
+		    self.dfs(root.left, result, num + str(root.val))
+	    if root.right:
+		    self.dfs(root.right, result, num + str(root.val))
+            
+            
+            #O(n)
+            #O(n)
```

Comment:
time complexity is O(N) which is correct
but space is O(Height of the tree) which will be equal to n in worst case